doc/releases/changelog-dev.md

```diff
@@ -29,6 +29,9 @@
   to be set as trainable), and therefore the derivatives are computed more efficiently.
   [(#3697)](https://github.com/PennyLaneAI/pennylane/pull/3697)
 
+* The `qml.equal` function now handles comparisons of `ParametrizedEvolution` operators.
```

Comment:
Not sure if this should be in the improvements or bug fixes section.

Reply:
Good point, bug fixes might make more sense!